### PR TITLE
Fixes #13058 caught exception thrown during view file rendering produces wrong output

### LINKED ISSUE #13058: Catched View exception still messes up page output
Migrated from https://github.com/yiisoft/yii2-swiftmailer/issues/38

### What steps will reproduce the problem?

1) Create web controller action:

```php
    public function actionRender()
    {
        try {
            return $this->render('exception');
        } catch (\Exception $e) {
            return $this->render('index');
        }
    }
```

2) Create view file `exception.php` which throws some exception inside:

```php
<h1>File Begin</h1>
<?php throw new Exception('test'); ?>
<h1>File End</h1>
```

### What is the expected result?

Fallback view rendered correctly

### What do you get instead?

Line `<h1>File Begin</h1>` from the view, which causes an exception appears at the top of the render result.

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.10
| PHP version      | 5.6
| Operating system | CentOS

### PR DESCRIPTION
**Pull request by klimov-paul**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #13058

### CODE REVIEW

File: framework/base/View.php [hunk 1]
```diff
@@ -322,12 +322,21 @@ public function afterRender($viewFile, $params, &$output)
...
         ob_start();
         ob_implicit_flush(false);
         extract($_params_, EXTR_OVERWRITE);
-        require($_file_);
-
-        return ob_get_clean();
+        try {
+            require($_file_);
+            return ob_get_clean();
+        } catch (\Exception $e) {
```
Comment (cebe) [inline framework/base/View.php]: for PHP 7 compatibility this should include catching \Throwable too.

**Review by cebe — CHANGES_REQUESTED** [review]
other than Throwable it looks good to me.

File: framework/base/View.php (hunk 1 above)
Comment (klimov-paul) [inline framework/base/View.php]: added

**Review by cebe — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**